src/main/java/gumtree/spoon/builder/NodeCreator.java

```diff
@@ -135,13 +135,20 @@ private void computeTreeOfTypeReferences(CtTypeReference<?> type, ITree parentTy
 			builder.addSiblingNode(returnType);
 		}
 
-		for (CtTypeReference thrown : e.getThrownTypes()) {
-			ITree thrownType = builder.createNode("THROWS", thrown.getQualifiedName());
-			thrownType.setMetadata(SpoonGumTreeBuilder.SPOON_OBJECT, thrown);
-			type.putMetadata(SpoonGumTreeBuilder.GUMTREE_NODE, thrownType);
-			builder.addSiblingNode(thrownType);
+		if (!e.getThrownTypes().isEmpty()) {
+			String typeLabel = "ThrownTypes";
+			ITree thrownTypeRoot = builder.createNode("THROWN_TYPES", typeLabel);
```

Comment:
for nodes with no label, it's better to put an empty label, the resulting tree is easier to understand and the diff is faster (never measured how much though)

Reply:
> the diff is faster (never measured how much though)

How does one measure execution time? By logging the start and end time before and after `new AstComparator().compare()`?

I will make this change and a similar change would be required in [one of my previous PR](https://github.com/SpoonLabs/gumtree-spoon-ast-diff/pull/197/files#diff-19ac4c8d46440718985d3a70cfd1e2eff828f1cdbeb99bc306993a5783e6b3abR100). I will submit a separate PR for the latter.

Reply:
> How does one measure execution time? By logging the start and end time before and after

That's the most naive way, but it is informative. More reliable measurements are done with frameworks such as JMH.

Reply:
> How does one measure execution time?

Actually, a better question would be how should I do benchmark testing.

Reply:
I don't think it's high priority now.

Reply:
Right but it was a really small change so I already submitted a PR for that - #200 .